test/cmd/diff.sh

```diff
@@ -41,9 +44,18 @@ run_kubectl_diff_tests() {
     output_message=$(kubectl diff -f hack/testdata/pod-changed.yaml || test $? -eq 1)
     kube::test::if_has_string "${output_message}" 'k8s.gcr.io/pause:3.0'
 
+    # Test found diff with server-side apply
+    output_message=$(kubectl diff -f hack/testdata/pod-changed.yaml --server-side || test $? -eq 1)
+    kube::test::if_has_string "${output_message}" 'k8s.gcr.io/pause:3.0'
+
     # Test that we have a return code bigger than 1 if there is an error when diffing
     kubectl diff -f hack/testdata/invalid-pod.yaml || test $? -gt 1
 
+    # Test empty diff with server-side apply
+    kubectl apply -f hack/testdata/pod.yaml --server-side --force-conflicts
+    # Make sure that diffing the resource right after returns nothing (0 exit code).
```

Comment:
We clearly need to change that in the future, maybe add a comment explaining why we're doing the test here. Also let's open an issue to track that we need to fix that when the apply bug is fixed.

Reply:
Ah ok, I'll update the comment.

Opened https://github.com/kubernetes/kubernetes/issues/88816

Reply:
Done.